### PR TITLE
Separate CSV from normal locktime

### PR DESCRIPTION
**Pull request by dcousens**
Motivation: https://github.com/bitcoin/bitcoin/pull/6564#discussion_r37251487

Untested (just building it now),  but,  should be enough to understand what I meant.

**edit**: now builds

### CODE REVIEW

File: src/script/interpreter.cpp [hunk 1]
```diff
@@ -1166,30 +1166,8 @@ bool TransactionSignatureChecker::CheckSig(const vector<unsigned char>& vchSigIn
     return true;
 }
 
-bool TransactionSignatureChecker::CheckLockTime(const CScriptNum& nLockTime, bool fSequence) const
+static bool SomeCheckLockTime(const CScriptNum& nLockTime, const int64_t txToLockTime)
```
Comment (dcousens) [inline src/script/interpreter.cpp:1169]: Probably want a better name,  and it doesn't need to be static either.

**Comment by dcousens** [comment]
ping @maaku

**Comment by btcdrak** [comment]
ping @maaku

**Comment by btcdrak** [comment]
I like this separation, it's more understandable.

**Pull request merged**